dnp3/src/outstation/session.rs

```diff
@@ -1052,6 +1229,55 @@ impl OutstationSession {
                         }
                     }
                 }
+                HeaderDetails::OneByteCount(_, CountVariation::Group50Var1(seq)) => {
+                    if let Some(value) = seq.single() {
+                        match self.application.write_absolute_time(value.time) {
+                            WriteTimeResult::NotSupported => {
+                                iin2 |= Iin2::NO_FUNC_CODE_SUPPORT;
+                            }
+                            WriteTimeResult::InvalidValue => {
+                                iin2 |= Iin2::PARAMETER_ERROR;
+                            }
+                            WriteTimeResult::Ok => (),
+                        }
+                    } else {
+                        tracing::warn!("request didn't have a single g50v1");
+                        iin2 |= Iin2::PARAMETER_ERROR;
+                    }
+                }
+                HeaderDetails::OneByteCount(_, CountVariation::Group50Var3(seq)) => {
+                    if let Some(value) = seq.single() {
```

Comment:
Let's unroll all these nested if/else statements somehow. This makes my brain hurt.